src/libs/actions/IOU.ts

```diff
@@ -480,6 +480,34 @@ type CreateTrackExpenseParams = {
     transactionParams: TrackExpenseTransactionParams;
 };
 
+type BuildOnyxDataForInvoiceParams = {
+    chat: {
+        chatReport: OnyxEntry<OnyxTypes.Report>;
+        chatCreatedAction: OptimisticCreatedReportAction;
+        reportPreviewAction: ReportAction;
+        isNewChatReport: boolean;
+    };
+    iou: {
+        iouCreatedAction: OptimisticCreatedReportAction;
+        iouAction: OptimisticIOUReportAction;
+        iouReport: OnyxTypes.Report;
+    };
+    transactionParams: {
+        transaction: OnyxTypes.Transaction;
+        transactionThreadReport: OptimisticChatReport;
+        transactionThreadCreatedReportAction: OptimisticCreatedReportAction | null;
+    };
+    policyParams: RequestMoneyPolicyParams;
+    optimisticData: {
+        optimisticRecentlyUsedCurrencies?: string[];
+        optimisticPolicyRecentlyUsedCategories: string[];
+        optimisticPolicyRecentlyUsedTags: OnyxTypes.RecentlyUsedTags;
+        optimisticPersonalDetailListAction: OnyxTypes.PersonalDetailsList;
+    };
+    companyName?: string;
+    companyWebsite?: string;
+};
```

Comment:
Looks good, but please remove the redundant prefixes in the sub-objects. For example, like we have here
https://github.com/Expensify/App/blob/78a4032032398bfa4f55f211ff3cd91cfae0521d/src/libs/actions/IOU.ts#L399-L403

Reply:
I think I prefer doing it without de-structuring `iou.createdAction.reportActionID` because then it's less redundant. However, I have allowed it with de-structuring before and it does have the advantage that you don't have to update all the places these variables are used.

If you would please update it to use dot notation I would appreciate that and I think it sets a good standard moving forward. Part of the goal of this refactor is to reduce the large number of variables to manage, and have a small number of object variables that are well organized and documented by their types.